lib/Service/AliasesService.php

```diff
@@ -79,4 +79,15 @@ public function delete($aliasId, $currentUserId) {
 		return $alias;
 	}
 
+	/**
+	 * Deletes all aliases of an account.
+	 * @param int $accountId the account which aliases will be deleted
+	 * @param string $currentUserId the user whom the account belongs to
+	 */
+	public function deleteAll($accountId, $currentUserId) {
+		$aliases = $this->mapper->findAll($accountId, $currentUserId);
+		foreach ($aliases as $alias) {
```

Comment:
Not quite efficient but fine for account deletions :wink:

Reply:
Agree, I wanted to avoid creating a new mapper method, but a dedicated SQL query that deletes all at once would be better.